workspaces/github-discussions/plugins/search-backend-module-github-discussions/src/collators/GithubDiscussionsCollatorFactory.ts

```diff
@@ -0,0 +1,245 @@
+
+  private constructor(options: GithubDiscussionsCollatorFactoryOptions) {
+    this.logger = options.logger.child({ documentType: this.type });
+    this.credentialsProvider = options.credentialsProvider;
+    this.githubIntegration = options.githubIntegration;
+    this.timeout = options.timeout;
+    this.url = options.url;
+    this.cacheBase = options.cacheBase;
+    this.clearCacheOnSuccess = options.clearCacheOnSuccess;
+    this.discussionsBatchSize = options.discussionsBatchSize;
+    this.commentsBatchSize = options.commentsBatchSize;
+    this.repliesBatchSize = options.repliesBatchSize;
+  }
+
+  static async fromConfig({
+    logger,
+    config,
+  }: GithubDiscussionsCollatorFactoryConstructorOptions) {
+    const _config = config.getConfig('search.collators.githubDiscussions');
+    const schedule = _config.has('schedule')
+      ? readSchedulerServiceTaskScheduleDefinitionFromConfig(
+          _config.getConfig('schedule'),
+        )
+      : DEFAULT_SCHEDULE;
+
+    const timeout = Duration.fromObject(schedule.timeout).as('milliseconds');
+    const integrations = ScmIntegrations.fromConfig(config);
+    const { github: githubIntegration } = integrations;
+    const credentialsProvider =
+      DefaultGithubCredentialsProvider.fromIntegrations(integrations);
+
+    const url = _config.getString('url');
+    const cacheBase = _config.getOptionalString('cacheBase');
+    const clearCacheOnSuccess = _config.getOptionalBoolean(
+      'clearCacheOnSuccess',
+    );
+    const discussionsBatchSize = _config.getOptionalNumber(
+      'discussionsBatchSize',
+    );
+    const commentsBatchSize = _config.getOptionalNumber('commentsBatchSize');
+    const repliesBatchSize = _config.getOptionalNumber('repliesBatchSize');
+
+    return new GithubDiscussionsCollatorFactory({
+      logger,
+      credentialsProvider,
+      githubIntegration,
+      timeout,
+      url,
+      cacheBase,
+      clearCacheOnSuccess,
+      discussionsBatchSize,
+      commentsBatchSize,
+      repliesBatchSize,
+    });
+  }
+
+  async getCollator() {
+    return Readable.from(this.execute());
+  }
+
+  async *execute(): AsyncGenerator<GithubDiscussionsDocument> {
+    const logger: typeof console = {
+      log: this.logger.info.bind(this.logger),
+      info: this.logger.info.bind(this.logger),
+      error: this.logger.error.bind(this.logger),
+      warn: this.logger.warn.bind(this.logger),
+      dir: this.logger.info.bind(this.logger),
+      debug: this.logger.debug.bind(this.logger),
+    } as unknown as typeof console;
+
+    const url = gh(this.url);
+    assert(url !== null, `Not parsable as a Github URL`);
+    const { name: repo, owner: org } = url;
+    assert(org !== null, `Discussions url is missing organization name`);
+    assert(repo !== null, `Discussion url is missing repository`);
+
+    const integration = this.githubIntegration.byUrl(url.href);
+
+    assert(integration, `Could not retrieve a Github integration for ${url}`);
```

Comment:
can we do all this checks in `fromConfig` instead? Is there any specific reason why you are using this `assert-ts` library instead of just throwing plain errors?